stripe/src/test/java/com/stripe/android/view/CountryAdapterTest.java

```diff
@@ -35,21 +35,36 @@ public void setup() {
 
         ActivityController<AddressInputTestActivity> activityController =
                 Robolectric.buildActivity(AddressInputTestActivity.class).create().start();
-        mCountryAdapter = new CountryAdapter(activityController.get());
+        List<String> countries = new ArrayList<>(CountryUtils.getCountryNameToCodeMap().keySet());
+        mCountryAdapter = new CountryAdapter(activityController.get(), countries);
     }
 
     @Test
     public void getOrderedSystemLocalesTest() {
-        List<Pair<String, String>> countryToDisplayNames = mCountryAdapter.getOrderedCountries();
         Locale currentLocale = mCountryAdapter.getCurrentLocale();
-        assertEquals(countryToDisplayNames.get(0).first, currentLocale.getCountry());
+        assertEquals(mCountryAdapter.getItem(0), currentLocale.getDisplayCountry());
         boolean ordered = true;
         // Skip the first comparision since we moved the current locale up
-        for (int i = 2; i < countryToDisplayNames.size(); i++) {
-            if (countryToDisplayNames.get(i).second.compareTo(countryToDisplayNames.get(i - 1).second) < 0) {
+        for (int i = 2; i < mCountryAdapter.getCount(); i++) {
+            if (mCountryAdapter.getItem(i).compareTo(mCountryAdapter.getItem(i-1)) < 0) {
                 ordered = false;
             }
         }
         assertTrue(ordered);
     }
+
+    @Test
+    public void filter_whenCountryInput_filtersCorrectly() {
```

Comment:
Might want to split this into the cases of when you filter nothing versus when you filter some stuff.